pkg/endpointslice/map.go

```diff
@@ -40,36 +68,86 @@ func NewMap() *Map {
 }
 
 func (m *Map) Put(es *discovery.EndpointSlice) {
-	key, ok := es.Labels[constants.LabelServiceImportName]
+	key, ok := getKey(es)
 	if !ok {
+		klog.V(log.DEBUG).Infof("Failed to get key labels from %#v", es.Labels)
 		return
 	}
 
-	epInfo := &endpointInfo{
-		key:     key,
-		hostIPs: make(map[string][]string),
+	cluster, ok := es.Labels[constants.LabelSourceCluster]
+
+	if !ok {
+		klog.Warningf("Cluster label missing on %#v", es.Labels)
```

Comment:
This is unexpected so I think it's fine to print the whole object.  Just printing the labels doesn't provide any identifying info. We need at least name and namespace as well. Same on L73.

Reply:
> This is unexpected so I think it's fine to print the whole object. Just printing the labels doesn't provide any identifying info. We need at least name and namespace as well. Same on L73.

Endpoints can go into thousands. How about printing ObjectMeta?